src/program.rs

```diff
@@ -207,6 +217,132 @@ where
     };
 }
 
+/// Create a markdown file with documentation for the specified FileId.
```

Comment:
Wouldn't functions from this point on be better in their own module, or at least submodule? They feel only tangentially related to `program` (which is, in itself, just a kind of interface bundling several moving parts of the codebase)